### PR TITLE
bpo-41028: Doc: Move switchers to docsbuild-scripts.

### PR DESCRIPTION
**Pull request by JulienPalard**
TL;DR: I'm moving language and version switchers (manually maintained in every branches), and "indexlayout" (left menu, in docs.python.org, with a list of versions, manually maintained in every branches), to docsbuild-scripts.

The current version of docs.python.org already uses the new language picker, version picker, and indexsidebar, I build all french versions with it manually, other languages will be built in a few hours.

There's no hurry to merge this PR, docsbuild-scripts accomodate to its presence or absence, and it does not need to be backported.

More info: https://mail.python.org/pipermail/doc-sig/2020-June/004200.html

I took some liberties:

- I added some translation to dummy.html, even if docsbuild-script don't use them yet, it should in the future. 
- I added an id to the `<li>` where the language and version switchers should land, to avoid ugly selectors to try to find it, and eventual bugs associated with ugly selectors. Docsbuild-script don't use it yet but should in the future.

This should avoid commits like: 19e3e00264, 85c6f8c65c, 3f36043db2, 59e7bbcaa4, which had to be backported as far as possible (I was not a big fan of pushing thoses changes to security-only branches, but it was the only way to have an up-to-date sidebar and switcher on their doc).

https://bugs.python.org/issue41028

### CODE REVIEW

**Comment by JulienPalard** [comment]
This goes as expected in production, all 3.7 translations are built, and all french translations are built, I see no issues at the moment, starting to build docs for 3.8.

**Comment by JulienPalard** [comment]
In production, this is now starting to fix links and picker for the 2.7 version, which was stating:
```
Docs by version

    Python 3.9 (in development)
```

and now displays:
```
Docs by version

    Python 3.10 (in development)
```

**Comment by JulienPalard** [comment]
Don't merge until we have a concensus on https://github.com/python/python-docs-theme/pull/46.

**Review by ned-deily — APPROVED** [review]
LGTM.  I think the "stand-alone" sidebar looks fine with the `Stable`, `Development`, and `All Versions` links.  This is a really nice maintainability improvement.

**Comment by JulienPalard** [comment]
Rebase on top of master, this is used since a few month in production and works, so I'll merge this soon.
